code-challenges.test.js

```diff
@@ -27,41 +27,110 @@ const numbersArray2 = [24, 27, 30, 33, 36]
+    // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+    const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+    // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+  expect(capWords(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+  expect(capWords(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+  })
+})
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
+const capWords = (words) => {
+  return words.map(word => word[0].toUpperCase() + word.slice(1))
+}
+
+
+// Pseudo code: 
+// I need a function called capWords that will take in words
+// have a return statement that maps the array
+// use .toUpperCase method to change each beginning letter using slice and the first index
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const vowelTester1 = "learn"
+describe("vowelFirst" ,() => {
+  it("takes in a string and logs the index of the first vowel", () => {
+    const vowelTester1 = "learn"
 // Expected output: 1
 const vowelTester2 = "academy"
 // Expected output: 0
 const vowelTester3 = "challenges"
 // Expected output: 2
+  expect(vowelFirst(vowelTester1)).toEqual([1])
+  expect(vowelFirst(vowelTester2)).toEqual([0])
+  expect(vowelFirst(vowelTester3)).toEqual([2])
+  })
+})
 
 // b) Create the function that makes the test pass.
+const vowelTester1 = "learn"
+// Expected output: 1
+const vowelTester2 = "academy"
+// Expected output: 0
+const vowelTester3 = "challenges"
+// Expected output: 2
+
+function vowelFirst(string) {
+  const vowels = "aeiouAEIOU"
+  const firstVowelIndex = Array.from(string).findIndex(char => vowels.includes(char))
+  return [firstVowelIndex]
+}
+
+const firstVowelIndex = vowelFirst(vowelTester1)
+console.log("The index of the first vowel is:", firstVowelIndex)
+
+const firstVowelIndex2 = vowelFirst(vowelTester2)
+console.log("The index of the first vowel is:", firstVowelIndex2)
+
+const firstVowelIndex3 = vowelFirst(vowelTester3)
+console.log("The index of the first vowel is:", firstVowelIndex3)
 
 // Pseudo code:
+// Make a function called vowelFirst that takes a string as an input
+// create a variable called vowels that will contain all the vowel characters
+// Now I have to convert all the vowel characters into an array of characters using array.from function
+// I have to usee the findindex method on the array to find the index of the first character that is included in the vowel string
+// Return an array containing the index of the first vowel found in the input string. Wrap the index in the square brackets to create an array.
+// (I honestly got stuck on this and had to research a lot of it. Even though I got it done, I still don't fully understand how I made this work)
```

Comment:
Remember that your function should output the expected output that is given by the question. Currently, your function does not pass the test. Here, you created a variable vowels that is a string of vowels. Then, you are creating an array from the string using Array.from, and then you are iterating through that array using findIndex. The built in method findIndex takes a callback function as an argument and will return the index of the first element in the array for which the callback function returns "true". The callback function will return true if the vowels string includes the current letter of iteration. So if the current letter of iteration is a vowel, it will be in the vowels string, and the callback function will return true, which will then make findIndex return the index of that letter.